mpz-core/src/prg.rs

```diff
@@ -0,0 +1,154 @@
+//! Implement AES-based PRG.
+
+use crate::{aes::AesEncryptor, Block};
+use rand::Rng;
+use rand_core::{
+    block::{BlockRng, BlockRngCore},
+    CryptoRng, RngCore, SeedableRng,
+};
+use std::mem;
+
+///Struct of PRG Core
+#[derive(Clone)]
+pub struct PrgCore {
+    aes: AesEncryptor,
+    state: u64,
+}
+
+
+// This implementation is somehow standard, and is adopted from Swanky.
+impl BlockRngCore for PrgCore {
+    type Item = u32;
+    type Results = [u32; 4 * AesEncryptor::AES_BLOCK_SIZE];
+
+    // Compute [AES(state)..AES(state+8)]
+    #[inline(always)]
+    fn generate(&mut self, results: &mut Self::Results) {
+        let states = [0; AesEncryptor::AES_BLOCK_SIZE].map(
+            #[inline(always)]
+            |_| {
+                let x = self.state;
+                self.state += 1;
+                Block::from(unsafe { mem::transmute::<_, [u8; 16]>([x, 0u64]) })
+            },
+        );
```

Comment:
We can avoid unsafe:

```suggestion
        let states: [Block; AesEncryptor::AES_BLOCK_SIZE] = 
            std::array::from_fn(|i| (i as u128).to_le_bytes().into()));
        self.state += AesEncryptor::AES_BLOCK_SIZE as u64;
```

Reply:
Just noticed this is wrong as we're not adding `self.state` to `i`, but the same jist